### ISSUE TITLE
Negative search.query_current stats confuses the Transport protocol

### ISSUE THREAD

**Issue by nomoa** [elastic/elasticsearch#25595]
We encoutered 2 nodes where the query.current stats are wrong (elastic 5.3.2):
```json
        "search": {
          "open_contexts": 0,
          "query_total": 615018,
          "query_time_in_millis": 1789208,
          "query_current": -2,
          "fetch_total": 612120,
          "fetch_time_in_millis": 701483,
          "fetch_current": 0,
          "scroll_total": 5,
          "scroll_time_in_millis": 4687558,
          "scroll_current": 0,
          "suggest_total": 0,
          "suggest_time_in_millis": 0,
          "suggest_current": 0
        },
```

Note `query_current: -2`

Getting such stats is only possible by calling directly the node that holds the invalid stats. Probably because negative values are not allowed as VLong: `Negative longs unsupported, use writeLong or writeZLong for negative numbers [-2]` but works when using XContentBuilder.
This makes a lot of the indices admin api unusable.

Note that the cluster was under heavy load with many rejections.
Restarting the nodes with broken stats solves the problem.

**Comment by jasontedor** [comment]
Closed by #24922

**Comment by jasontedor** [comment]
This duplicates a previous issue and is fixed in 5.4.1. You can see the fix by upgrading to this or a later version.
